### PR TITLE
Fix delayed repainting (and a few compile-time warnings)

### PR DESCRIPTION
**Pull request by scymtym**
* Fix incorrect behavior (and compile-time warnings) due to the `*inhibit-dispatch-repaint*` variable and associated mechanism

* Fix a few unused variable warnings and the incorrect `(setf stream-pointer-position)` method for `encapsulating-stream`

### CODE REVIEW

File: Core/clim-basic/repaint.lisp [hunk 1]
```diff
@@ -26,23 +26,55 @@
...
+          ;; DELAYED-REPAINTS.
+          (let ((*delayed-repaints* delayed-repaints))
+            (funcall thunk))
+          ;; Merge and execute recorded repaint requests.
+          (maphash (lambda (sheet regions)
+                     (dispatch-repaint sheet (reduce #'region-union regions)))
+                   delayed-repaints)))))
+
+(defmacro with-inhibited-dispatch-repaint (() &body body)
+  `(call-with-inhibited-dispatch-repaint (lambda () ,@body)))
```
Comment (dkochmanski) [inline Core/clim-basic/repaint.lisp]: convention used in mcclim is `invoke-with-whatever-..` and macro constructed with flet:

```
(defmacro with-foo (() &body body)
  (with-gensym (cont)
    `(flet ((,cont () ,@body))
      (declare (dynamic-extent #',cont))
      (invoke-with-foo #',cont)))
```

File: Core/clim-basic/repaint.lisp [hunk 2]
```diff
@@ -26,23 +26,55 @@
...
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;
 ;;; Repaint protocol functions.
 
-(defmethod dispatch-repaint ((sheet graft) region)
+#+no (defmethod dispatch-repaint :around ((sheet graft) region)
   (declare (ignore sheet region)))
```
Comment (dkochmanski) [inline Core/clim-basic/repaint.lisp]: I suppose this `#+no` means, that the method definition is meant to be removed?

File: Core/clim-basic/repaint.lisp (hunk 1 above)
Comment (scymtym) [reply Core/clim-basic/repaint.lisp]: Fixed.

File: Core/clim-basic/repaint.lisp (hunk 2 above)
Comment (scymtym) [reply Core/clim-basic/repaint.lisp]: Removed.

**Pull request merged**